src/crate/client/cursor.py

```diff
@@ -49,8 +50,45 @@ def execute(self, sql, parameters=None, bulk_parameters=None):
 
         self._result = self.connection.client.sql(sql, parameters,
                                                   bulk_parameters)
+
         if "rows" in self._result:
-            self.rows = iter(self._result["rows"])
+            transformed_result = False
+            if "col_types" in self._result:
+                transformed_result = True
+                self.rows = self.result_set_transformed()
+
+            if not transformed_result:
+                self.rows = iter(self._result["rows"])
+
+    def result_set_transformed(self):
+        """
+        Generator that iterates over each row from the result set
+        """
+        rows_to_convert = [True if col_type == 11 or col_type == 15 else False for col_type in
+                           self._result["col_types"]]
+        for row in self._result["rows"]:
+            gen_flags = (flag for flag in rows_to_convert)
+            yield [t_row for t_row in self._transform_date_columns(row, gen_flags)]
+
+    @staticmethod
+    def _transform_date_columns(row, gen_flags):
+        """
+        Generates iterates over each value from a row and converts timestamps to pandas TIMESTAMP
```

Comment:
Can we do something more elegant with [lambdas](https://www.geeksforgeeks.org/python-iterating-with-python-lambda/ ) ? (I have basic python knowledge though).

Reply:
I think using the generator here is fine. Let me know if you think otherwise.

Reply:
When looking at this again, specifically after cleaning up a bit with 6522e430284, I tend to agree with you. It is probably enough that `_transform_result_types`, which iterates all records, is a generator. On the other hand, `_transform_datetime_columns`, which iterates all columns per record, does not necessarily have to be a generator.

I will keep the original patch unchanged on this matter yet, and will await the official review.